src/librustc_mir/transform/const_prop.rs

```diff
@@ -985,8 +1026,48 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
             | TerminatorKind::GeneratorDrop
             | TerminatorKind::FalseEdges { .. }
             | TerminatorKind::FalseUnwind { .. } => {}
-            //FIXME(wesleywiser) Call does have Operands that could be const-propagated
-            TerminatorKind::Call { .. } => {}
+            TerminatorKind::Call { ref mut args, .. } => {
+                // Every argument in our function calls can be const propagated.
+                // We try to, before exiting.
+                for opr in args {
+                    // We see if the operand can be evaluated, and if so, we continue.
+                    if let Some(op_ty) = self.eval_operand(&opr, source_info) {
```

Comment:
I wonder if there's something slightly simpler that might perform better than engaging all of the miri machinery. I believe in the general case, we always spill arguments to a function call into temporaries. Which means that most all function calls look like `func(move _tmp0, move _tmp1, ...)`. We could probably simplify all of this logic just by doing something like

```rust
if let Some(l) = opr.place.and_then(|p| p.as_local()) {
    if let Some(value) = self.get_const(l) {
        if self.should_const_prop(value) {
            if let interpret::Operand::Immediate(interpret::Immediate::Scalar(scalar_mu)) = *value {
                *opr = self.operand_from_scalar(scalar, value.layout.ty, source_info.span);
            }
        }
    }
}
```

Reply:
> I believe in the general case, we always spill arguments to a function call into temporaries.

Can we measure that somehow? If we could only cover that particular case (`LocalKind::Temp` where the `Operand` is of the `Immediate::Scalar` variety) without losing too many actual cases, that would help by not having to invoke `eval_operand` at all.

PS: if we were to only implement one case on purpose, we definitely must document why. So that people in the future don't try to expand it without knowing this :3

Reply:
> Can we measure that somehow?

I believe the only case where we don't do this is when we are passing a pointer to an unsized variable:

https://github.com/rust-lang/rust/blob/0862458dad90a0d80827e22e3f86e33add6d847c/src/librustc_mir_build/build/expr/as_operand.rs#L119-L197

So 99.99% of the time, the simpler logic should work 🙂

Reply:
@wesleywiser I want us to have this reasoning written down next to the implementation of `const-prop` for function arguments, so that future people don't try to expand it without at least knowing why we don't have it there yet. 

Can I ask you to write the reasoning down? Because I don't quite understand it myself. 
I'll make sure it sits right next to the code, so don't worry about that part :)

Reply:
I left a comment about this on #71697.

BTW, I think since you're working on the last bit in that PR, we should probably close this PR since it won't be merged as-is.

Reply:
Yup! By closing this one point I think we can finally close it :)